app/uk/gov/hmrc/tariffclassificationfrontend/controllers/CancelRulingController.scala

```diff
@@ -42,10 +44,7 @@ class CancelRulingController @Inject()(verify: RequestActions,
 
   override protected val config: AppConfig = appConfig
   override protected val caseService: CasesService = casesService
-
-  override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = {
-    c.status == COMPLETED && (c.hasLiveRuling || c.application.isLiabilityOrder)
-  }
+  private implicit val clock: Clock = appConfig.clock
```

Comment:
What's the clock for?

Reply:
For the hasLiveRuling logic, permissions need to supply a clock somehow